jetson/percep/pcl.cpp

```diff
@@ -186,177 +186,81 @@ void PCL::FindInterestPoints(std::vector<pcl::PointIndices> &cluster_indices,
+        compareLine(0,-HALF_ROVER), compareLine(0,HALF_ROVER))){
+            cerr << "CENTER PATH IS CLEAR!!!" << std::endl;
         return 0;
     }
-    
+
     //Initialize base cases outside of scope
     vector<int> centerObstacles = {obstacles.at(0), obstacles.at(1)};
-    double newAngle = 0;
-    double newSlope = 0;
-
-    // create left and right angle vlaue that default at 360 degrees
-    double leftAngle = 360;
-    double rightAngle = 360;
-
-    //If Center Path is blocked check left until have to turn too far
-    while(newAngle > -70 && newSlope <= 0){
-        
-        //Declare Stuff
-        double x = pt_cloud_ptr->points[obstacles.at(0)].x;
-        double z = pt_cloud_ptr->points[obstacles.at(0)].z;
-        distance = z;
-        double zOffset = 0;
-        double xOffset = 0;
-        
-        if(x <= 0){
-            //Calculate angle from X axis, slope of line, and length of line to left obstacle
-            double slope = x/z;
-            double length = sqrt(x*x+z*z);
-            double angleOffset = acos(z/length)*180/PI;
-
-            //Calculate angle from X axis and length of line that is half rover distance from obstacle
-            double interior = asin((HALF_ROVER+buffer)/length)*180/PI;
-            double adjacent = 90-fabs(interior)-fabs(angleOffset);
-            double key = 90-fabs(adjacent);
-            zOffset = sin(key*PI/180)*(HALF_ROVER+buffer);
-            xOffset = cos(key*PI/180)*(HALF_ROVER+buffer);
-        }
-
-        else{
-            //Calculate angle from X axis, slope of line, and length of line to left obstacle
-            double slope = x/z;
-            double length = sqrt(x*x+z*z);
-            double angleOffset = acos(z/length)*180/PI;
-
-            //Calculate angle new point will be from X axis
-            double interior = asin((HALF_ROVER+buffer)/length)*180/PI;
-            double key = 90-(fabs(interior)-fabs(angleOffset));
-            xOffset = sin(key*PI/180)*(HALF_ROVER+buffer);
-            zOffset = cos(key*PI/180)*(HALF_ROVER+buffer);
-        }
-
-        //Find points on line of new path
-        double newPathz = z-fabs(zOffset);
-        double newPathx = x-fabs(xOffset);
-        newSlope = newPathx/newPathz;
-        newAngle = atan(newPathx/newPathz)*180/PI;
-
-        //Create left and right lines based on new path
-        double shiftX = HALF_ROVER/sin((90-newAngle)*PI/180);
-
-        compareLine leftLine (newSlope,-shiftX);
-        compareLine rightLine (newSlope, shiftX);
 
-        obstacles.clear();
-        if(CheckPath(interest_points, viewer, obstacles, leftLine, rightLine))
-        {
-            #if PERCEPTION_DEBUG
-            std::cout<<"!!!!!!!!!!!!!!!!!!!!!!!!!!!!!FOUND NEW PATH AT: "<<newAngle<<std::endl;
-            #endif
-            leftAngle = newAngle;
-            break;
-        }
-            
-    }
+    //Find Clear left path
+    leftAngle = getAngle(10, 0, interest_points, viewer, obstacles);
```

Comment:
What do 10 and 0 mean?

Reply:
10 is the buffer and 0 is the direction. The direction is explained in the header of the getAngleOffCenter function. The 10 is just a magic number